src/Libraries/CoreNodeModels/Input/CustomSelection.cs

```diff
@@ -0,0 +1,313 @@
+using Autodesk.DesignScript.Runtime;
+using Dynamo.Graph;
+using Dynamo.Graph.Nodes;
+using Newtonsoft.Json;
+using ProtoCore.AST.AssociativeAST;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Windows.Input;
+using System.Xml;
+
+
+namespace CoreNodeModels.Input
+{
+    /// <summary>
+    /// This node allow the user to create a dropdown menu with an with an arbitrary number of customization items
+    /// </summary>
+    [NodeName("Custom Dropdown Menu")]
```

Comment:
@LongNguyenP @saintentropy - not sure if this is worth doing, but perhaps this node should actually serialize as the `cogs` `Selection` input type by overriding `InputData` ?